operator/v1alpha1/types_kueue.go

```diff
@@ -0,0 +1,174 @@
+	// only manage the workloads that are specfied in this list.
+	// This is a required field.
+	// +required
+	Integrations Integrations `json:"integrations"`
+}
+
+// KueueStatus defines the observed state of Kueue
+type KueueStatus struct {
+	operatorv1.OperatorStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// KueueList contains a list of Kueue
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type KueueList struct {
+	metav1.TypeMeta `json:",inline"`
+	// metadata for the list
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+	// items is a slice of Kueue
+	// this is a cluster scoped resource and there can only be 1 Kueue
+	// +kubebuilder:validation:MaxItems=1
+	// +required
+	Items []Kueue `json:"items"`
+}
+
+// +kubebuilder:validation:Enum=BatchJob;RayJob;RayCluster;JobSet;MPIJob;PaddleJob;PytorchJob;TFJob;XGBoostJob;AppWrappers;Pod;Deployment;StatefulSet;LeaderWorkerSet
+type KueueIntegrations string
+
+const (
+	BatchJob        KueueIntegrations = "BatchJob"
+	RayJob          KueueIntegrations = "RayJob"
+	RayCluster      KueueIntegrations = "RayCluster"
+	JobSet          KueueIntegrations = "JobSet"
+	MPIJob          KueueIntegrations = "MPIJob"
+	PaddeJob        KueueIntegrations = "PaddeJob"
+	PyTorchJob      KueueIntegrations = "PyTorchJob"
+	TFJob           KueueIntegrations = "TFJob"
+	XGBoostJob      KueueIntegrations = "XGBoostJob"
+	AppWrappers     KueueIntegrations = "AppWrappers"
+	Pod             KueueIntegrations = "Pod"
+	Deployment      KueueIntegrations = "Deployment"
+	StatefulSet     KueueIntegrations = "StatefulSet"
+	LeaderWorkerSet KueueIntegrations = "LeaderWorkerSet"
+)
+
+// This is the GVK for an external framework.
+// Controller runtime requires this in this format
+// for api discoverability.
+type ExternalFramework struct {
+	// group of externalFramework
+	// must be a valid qualified name consisting of a lower-case alphanumeric string,
+	// and hyphens of at most 63 characters in length. The name must start and end with an alphanumeric character.
+	// The name may be optionally prefixed with a subdomain consisting of lower-case alphanumeric characters,
+	// hyphens and periods, of at most 253 characters in length.
+	// Each period separated segment within the subdomain must start and end with an alphanumeric character.
+	// The optional prefix and the name are separate by a forward slash (/).
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:XValidation:rule="self.size() == 0 || !format.qualifiedName().validate(self).hasValue()",message="a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character."
```

Comment:
I think we probably want the dns1123Subdomain validation instead of the qualifiedName validation:
```suggestion
	// +kubebuilder:validation:XValidation:rule="self.size() == 0 || !format.dns1123Subdomain().validate(self).hasValue()",message="a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character."
```

Reply:
done.